src/libs/actions/Report.js

```diff
@@ -1084,6 +1086,36 @@ function deleteReportComment(reportID, reportAction) {
                 };
 
                 Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, reportActionsToMerge);
+            } else {
+                // Update last message information in the report object
+                Onyx.connect({
+                    key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
+                    callback: _.once((reportActionList) => {
+                        // Find last non-empty message
+                        const lastReportAction = !_.isEmpty(reportActionList)
+                            ? _.find(Object.values(reportActionList).reverse(),
+                                ({message, actionName}) => _.last(message).html !== ''
+                                && actionName !== 'CREATED')
+                            : null;
+
+                        // Populate last message details
+                        const lastMessageDetails = {};
+                        if (lastReportAction) {
+                            // Replace line breaks with spaces and remove all HTML from the message
+                            lastMessageDetails.lastMessageText = lastReportAction.isAttachment
+                                ? '[Attachment]'
+                                : Str.stripHTML(_.last(lastReportAction.message).html.replace(/((<br[^>]*>)+)/gi, ' '));
+                            lastMessageDetails.lastMessageTimestamp = lastReportAction.timestamp;
+                            lastMessageDetails.lastActorEmail = lodashGet(lastReportAction, 'actorEmail', '');
+                        } else {
+                            lastMessageDetails.lastMessageText = '';
+                            lastMessageDetails.lastMessageTimestamp = 0;
+                            lastMessageDetails.lastActorEmail = '';
+                        }
+
+                        Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, lastMessageDetails);
```

Comment:
This statement causes a recursion.

Reply:
Can you please explain how does it cause a recursion? Is there some specific edge case I didn't see. The code runs only once after the response from BE arrives.

Reply:
Oh sorry didn't notice at first sight that this is a different subscription.